File: LeetCode/3424-minimum-cost-to-make-arrays-identical/3424-minimum-cost-to-make-arrays-identical.cpp
class Solution {
public:
    #define all(x) x.begin(), x.end()
    #define ll long long
    long long minCost(vector<int>& arr, vector<int>& brr, long long k) {
        if (arr == brr) return 0;
        int n = arr.size();
        if (n == 1) return abs(arr[0] - brr[0]);
        ll ans2 = 0;
        for (ll i = 0; i < n; i++) {
            ans2 += abs(arr[i] - brr[i]);
        }
        sort(all(arr));
        sort(all(brr));
        ll ans1 = k;        
        for (ll i = 0; i < n; i++) {
            ans1 += abs(arr[i] - brr[i]);
        }
        return min(ans1, ans2);
    }
};

File: LeetCode/3424-minimum-cost-to-make-arrays-identical/3424-minimum-cost-to-make-arrays-identical.java
class Solution {
    public long minCost(int[] arr, int[] brr, long k) {
        if (Arrays.equals(arr, brr)) return 0;
        int n = arr.length;
        if (n == 1) return Math.abs(arr[0] - brr[0]);
        long ans2 = 0;
        for (int i = 0; i < n; i++) {
            ans2 += Math.abs(arr[i] - brr[i]);
        }
        Arrays.sort(arr);
        Arrays.sort(brr);
        long ans1 = k;
        for (int i = 0; i < n; i++) {
            ans1 += Math.abs(arr[i] - brr[i]);
        }
        return Math.min(ans1, ans2);
    }
}